torch/csrc/jit/serialization/python_print.cpp

```diff
@@ -753,15 +753,21 @@ struct PythonPrintImpl {
       if (version_entry != get_operator_version_map().end()) {
         const auto& entry = version_entry->second;
         // TODO (tugsuu) move this calculation into a seperate step.
-        min_version_ = std::max(
-            min_version_, uint64_t(entry[entry.size() - 1].bumped_at_version));
+        uint64_t current_version = entry[entry.size() - 1].bumped_at_version;
+        uint64_t historic_version = get_min_version_for_kind(node->kind());
+
+        // True means we solely calculate based on upgrader version
+        if (get_version_calculator_flag()) {
+          min_version_ = std::max(min_version_, current_version);
+        } else {
```

Comment:
I think we briefly discussed about this. Should we error out here if `min_version` and `historic_version` do not match? Since this error happens at compile/save time, it is a good place to detect issues. Also provide a way to override this check and use legacy versioning method

Reply:
Hmm it seems weird to error out based on that because this is an implementation detail. And this check would only work for 'div' and 'full' because historic (legacy) map doesn't have the other ops.

Reply:
Good point, it is up to you. I was just thinking about it as a temporary measure to make sure the migration is absolutely safe.